sample/transforms.go

```diff
@@ -0,0 +1,122 @@
+package sample
+
+import (
+	"cmp"
+	"math"
+	"slices"
+
+	pq "github.com/emirpasic/gods/v2/queues/priorityqueue"
+)
+
+type Transform interface {
+	Apply([]float64) []float64
```

Comment:
This doesn't need to be public anymore

Reply:
Will be addressed in follow-up

Reply:
If this is only used in the `sample` package, no harm in lowercasing it in this PR